src/instancemanager.jl

```diff
@@ -207,14 +207,14 @@ function MOI.canmodifyconstraint(m::InstanceManager, cindex::CI, change)
     return true
 end
 
-function MOI.modifyconstraint!(m::InstanceManager, cindex::CI{F,S}, func::F) where {F<:MOI.AbstractFunction, S<:MOI.AbstractSet}
-    if m.mode == Automatic && m.state == AttachedSolver && !MOI.canmodifyconstraint(m.solver, cindex, typeof(func))
+function MOI.modifyconstraint!(m::InstanceManager, cindex::CI{F}, change::Union{F, MOI.AbstractFunctionModification}) where F<:MOI.AbstractFunction
```

Comment:
This is a confusing method signature. What happens to `F` when `change` is a modification?

Reply:
`F` is determined by the type of `cindex`. Then `change` is either `F` or a function modification

Reply:
The extra complexity in the type signature is probably not worth the ability to get a method error for the one corner case where `F` doesn't match the type of the function. We already check this and give a potentially more useful error message with `canmodifyconstraint`.

Reply:
I have simplified it